### PR TITLE
Add schema_url fields as described in OTEP 0152

### PR DESCRIPTION
**Pull request by tigrannajaryan**
This adds schema_url fields introduced in OTEP 0152 (Telemetry Schemas):
https://github.com/open-telemetry/oteps/blob/main/text/0152-telemetry-schemas.md#otlp-changes

### CODE REVIEW

**Review by jmacd — APPROVED** [review]

**Review by jmacd — APPROVED** [review]
Looks good!

File: opentelemetry/proto/logs/v1/logs.proto
Comment (bogdandrutu) [inline opentelemetry/proto/logs/v1/logs.proto:54]: How does this impact the API? Why is it not part of the `InstrumentationLibrary`?

File: opentelemetry/proto/logs/v1/logs.proto
Comment (bogdandrutu) [inline opentelemetry/proto/logs/v1/logs.proto:39]: Can we start without this optimization? How is this set in the API/SDK?

Comment (tigrannajaryan) [reply opentelemetry/proto/logs/v1/logs.proto:39]: See application-wide schema URL in SDK: https://github.com/open-telemetry/opentelemetry-specification/pull/1666/files

Comment (tigrannajaryan) [reply opentelemetry/proto/logs/v1/logs.proto:39]: This is not an optimization. It is _required_ to describe the attributes of the Resource.

**Review by gramidt — APPROVED** [review]
Looks good!

**Review by jsuereth — APPROVED** [review]

**Comment by tigrannajaryan** [comment]
I have just chatted with @bogdandrutu and he had some ideas that may result in putting schema_url inside InstrumentationLibrary message instead of InstrumentationLibraryXXX message (no difference in semantics).
Please don't merge this yet, while I am thinking about it.

File: opentelemetry/proto/logs/v1/logs.proto
Comment (tigrannajaryan) [reply opentelemetry/proto/logs/v1/logs.proto:54]: The API impact is shown here https://github.com/open-telemetry/opentelemetry-specification/pull/1666

I would prefer to keep `schema_url` inside InstrumentationLibrarySpans since it is not a property of the instrumentation library, it is a “meta” field which describes other fields in InstrumentationLibrarySpans and I think putting it inside InstrumentationLibrary is obscure. It would also look a bit strange to use a field in message A (InstrumentationLibrary) to describe fields in message B (repeated Span), when A and B are siblings and do not have parent-child relationship.

File: opentelemetry/proto/logs/v1/logs.proto
Comment (tigrannajaryan) [reply opentelemetry/proto/logs/v1/logs.proto:39]: After discussing with @bogdandrutu and putting some more though into this I am going to make a change here. The `schema_url` here will describe the Resource, but will not have any effect on the spans (the overriding logic that it has in the current form). This is unnecessary complication, it saves maybe a tiny bit of data in a request, but creates confusion.
I am going to delete this "overriding" logic and simplify it. Will post a commit in a bit.

Comment (tigrannajaryan) [reply opentelemetry/proto/logs/v1/logs.proto:39]: OK, see the new commit https://github.com/open-telemetry/opentelemetry-proto/commit/dd92c019a527e55d1f7b926ed4618de7ef4a9369 that does this. Only the comments are changed (which changes the semantic of this field).
I believe this is simpler and easier to understand.

File: opentelemetry/proto/logs/v1/logs.proto
Comment (tigrannajaryan) [reply opentelemetry/proto/logs/v1/logs.proto]: Discussed with @bogdandrutu one more time and I believe what we have is the right approach. Resolving.

**Comment by tigrannajaryan** [comment]
Reviewers, please have another look. I will keep this open for 2 days just in case there are objections to the last change.

**Review by Aneurysm9 — APPROVED** [review]

**Review by jsuereth — APPROVED** [review]

**Comment by jmacd** [comment]
@tigrannajaryan Do you want this to merge and release in v0.9? The metrics group would like to release v0.9 very soon. Thanks!

**Comment by tigrannajaryan** [comment]
We have the necessary number of approvals and all comments are resolved, so merging.

**Pull request merged**
